enhancements/cluster-logging/cluster-logging-v2-apis.md

```diff
@@ -0,0 +1,565 @@
+
+### Non-Goals
+
+* "One click" deployment of a full logging stack as provided by **ClusterLogging** v1
+* Complete backwards compatibility to **ClusterLogForwarder** v1
+
+
+## Proposal
+
+
+### Workflow Description
+
+The following workflow describes the first user story which is a superset of the others and allows deployment of a full logging stack to collect and forward logs to a Red Hat managed log store.
+
+**cluster administrator** is a human responsible for:
+
+* Managing and deploying day 2 operators
+* Managing and deploying an on-cluster LokiStack
+* Managing and deploying a cluster-wide log forwarder
+
+**cluster-observability-operator** is an operator responsible for:
+
+* managing and deploying observability operands (e.g. LokiStack, ClusterLogForwarder, Tracing) and console plugins (e.g console-logging-plugin)
+
+**loki-operator** is an operator responsible for managing a loki stack.
+
+**cluster-logging-operator** is an operator responsible for managing log collection and forwarding.
+
+The cluster administrator does the following:
+
+1. Deploys the Red Hat **cluster-observability-operator**
+1. Deploys the Red Hat **loki-operator**
+1. Deploys an instance of **LokiStack** in the `openshift-logging` namespace
+1. Deploys the Red Hat **cluster-logging-operator**
+1. Creates a **ClusterLogForwarder** custom resource for the **LokiStack**
+
+The **cluster-observability-operator**:
+1. Deploys the console-logging-plugin for reading logs in the OpenShift console
+
+The **loki-operator**:
+1. Deploys the **LokiStack** for storing logs on-cluster
+
+The **cluster-logging-operator**:
+
+1. Deploys the log collector to forward logs to log storage in the `openshift-logging` namespace
+
+### API Extensions
+
+This API defines the following opinionated input sources which is a continuation of prior cluster logging versions:
+
+* **application**: Logs of container workloads running in all namespaces except **default**, **openshift***, and **kube***
+* **infrastructure**: journald logs from OpenShift nodes and container workloads running only in namespaces **default**, **openshift***, and **kube***
+* **audit**: The logs from OpenShift nodes written to the node filesystem by: Kubernetes API server, OpenShift API server, Auditd, and OpenShift Virtual Network (OVN).
+
+These are **reserved** words that represent input sources that can be referenced by a pipeline without an explicit input specification.
+
+More explicit specification of **audit** and **infrastructure** logs is allowed by creating a named input of that type and specifiying at least one of the allowed sources.
+
+#### ClusterLogForwarer CustomResourceDefinition:
+
+Following defines the next version of a ClusterLogForwarder. **Note:** The next version of this resources is part of a new API group to align log collection with
+the objectives of Red Hat observability.
```

Comment:
According to the [implementation PR](https://github.com/openshift/cluster-logging-operator/pull/2451/files#diff-1b76f67180e8ef3749f2ff1eb250d9f0a79ed20ce720505a88c4e867afc6bc02R16), this resource is meant to be `namespaced`. Can we outline the case when and how this resource is to be used on namespaces other than `openshift-logging`?